### PR TITLE
[ml] Add 'max_instance_count' to JobResourceConfiguration for supporting PyTorch Elastic jobs

### PR DESCRIPTION
**Pull request by svaruag**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import logging
 from typing import Any, Dict, Optional
 
-from azure.ai.ml._restclient.v2022_10_01.models import ResourceConfiguration as RestResourceConfiguration
+from azure.ai.ml._restclient.v2023_04_01_preview.models import ResourceConfiguration as RestResourceConfiguration
```
Comment (svaruag) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:9]: @luigiw QQ: 'max_instance_count' appears in `v2023_04_01_preview.models.ResourceConfiguration` why don't we have it in the non-preview package? (v2023_04_01.models.ResourceConfiguration). Is it ok to import from this here?

Comment (luigiw) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:9]: JobOperations switched to April preview. https://github.com/Azure/azure-sdk-for-python/blob/3f3c1eee5241f54836cccad2340e5af7352a4b72/sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py#L336 

Let's switch all job imports to April preview.

Comment (svaruag) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:9]: You mean replace `v2022_10_01` with `v2023_04_01_preview` across the board? There's about 71 files in all.

Comment (svaruag) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:9]: @luigiw please review again

Comment (svaruag) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:9]: (most of the job related imports were updated, but i guess there were some that were missed)

**Review by needuv — COMMENTED** [review]
you can revert the changes to `BatchDeployment`

**Comment by svaruag** [comment]
> you can revert the changes to `BatchDeployment`

Done. Also, added a hasattr check as discussed.

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_job/automl.py [hunk 2]
```diff
@@ -3,9 +3,10 @@
 # ---------------------------------------------------------
 from enum import Enum
 
-# pylint: disable=unused-import
-from azure.ai.ml._restclient.v2022_10_01_preview.models import NlpLearningRateScheduler
 from azure.ai.ml._restclient.v2023_02_01_preview.models import TrainingMode
```
[the remark this answers is not in the document]
Comment (svaruag) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_job/automl.py:6]: thanks, fixing it since i have to push a few more changes after review with Neehar

**Review by needuv — APPROVED** [review]

**Pull request merged**